### PR TITLE
removing information on mandate to clone go in GOPATH

### PR DESCRIPTION
**Pull request by girishramnani**
**What type of PR is this?**
> /kind documentation
> [skip ci]

**What does does this PR do / why we need it**:
we dont necessarily need to clone odo in GOPATH so changing that

**Which issue(s) this PR fixes**:

No issue

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [X] Documentation 

- [X] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: docs/dev/development.adoc [hunk 1]
```diff
@@ -29,7 +29,8 @@ First setup your fork of the odo project, following the steps below
 
  . Clone your fork:
 +
-NOTE: The following commands assume that you have the `$GOPATH` environment variable properly set. We highly recommend you place `odo` code into the $GOPATH.
+NOTE: odo uses `go modules` to manage dependencies which means you can clone the code anywhere you like but for backwards compatibility
```
Comment (dharmit) [inline docs/dev/development.adoc:32]: I thought it was enough to be using go.mod. What's the backwards compatibility that we're talking about here?

**Review by dharmit — COMMENTED** [review]
Just a question.

File: docs/dev/development.adoc (hunk 1 above)
Comment (girishramnani) [reply docs/dev/development.adoc:32]: If I remember correctly - go version 1.13 introduced go mod - if someone is using older version of golang then they won’t have access to golang modules

**Review by dharmit — APPROVED** [review]
/lgtm

**Pull request merged**